doc/POINTS_COORDINATES.md

```diff
@@ -0,0 +1,169 @@
+by scale and origin.
+
+The most precise coordinates are **map square** (ms) coordinates.  These refer to
+the tiles you see normally when playing the game.
+
+Two origins for map square coordinates are common:
+* **Absolute** coordinates, which are a global system for the whole game,
+  relative to a fixed origin.
+* **Local** coordinates, which are relative to the corner of the current "reality
+  bubble", or `map` roughly centered on the avatar.  In local map square
+  coordinates, `x` and `y` values will both fall in the range [0,120).
+
+The next scale is **submap** (sm) coordinates.  One submap is 24x24
+(`SEEX`x`SEEY`) maps squares.  Submaps are the scale at which chunks of the map
+are loaded or saved as they enter or leave the reality bubble.
+
+Next comes **overmap terrain** (omt) coordinates.  One overmap terrain is 2x2
+submaps.  Overmap terrains correspond to a single tile on the map view in-game,
+and are the scale of chunk of mapgen.
+
+Largest are **overmap** (om) coordinates.  One overmap is 180x180
+(`OMAPX`x`OMAPY`) overmap terrains.  Large-scale mapgen (e.g. city layout)
+happens one overmap at a time.
+
+Lastly, these is a system called **segment** (seg) coordinates.  These are only
+used in saving/loading submaps and you are unlikely to encounter them.
+
+As well as absolute and local coordinates, sometimes we need to use coordinates
+relative so some larger scale.  For example, when performing mapgen for a
+single overmap, we want to work with coordinates within that overmap.  This
+will be an overmap terrain-scale point relative to the corner of its containing
+overmap, and so typically take `x` and `y` values in the range [0,180).
+
+## Vertical coordinates
+
+Although `x` and `y` coordinates work at all these various scales, `z`
+coordinates are consistent across all contexts.  They lie in the range
+[-`OVERMAP_DEPTH`,`OVERMAP_HEIGHT`].
+
+## Vehicle coordinates
+
+TODO.
+
+## Point types
+
+To work with these coordinate systems we have a variety of types.  These are
+defined in [`coordinates.h`](../src/coordinates.h).  For example, we have
+`point_abs_ms` for absolute map-square coordinates.  The three parts of the
+type name are *dimension*`_`*origin*`_`*scale*.
+
+* **dimension** is either `point` for two-dimensional or `tripoint` for
+  three-dimensional.
+* **origin** specifies what the value is relative to, and can be:
+  * `rel` means relative to some arbitrary point.  This is the result of
+    subtracting two points with a common origin.  It would be used for example
+    to represent the offset between the avatar and a monster they are shooting
+    at.
+  * `abs` means global absolute coordinates.
+  * `sm` means relative to a corner of a submap.
+  * `omt` means relative to a corner of an overmap terrain.
+  * `om` means relative to a corner of an overmap.
+* **scale** means the scale as discussed above.
+  * `ms` for map square.
```

Comment:
* `mnt` for vehicle mount.
  * `veh` for translated vehicle.

Reply:
I added `mnt`, but (IMO) translated vehicle coordinates should use the `ms` scale, just with a different origin.

Reply:
works for me.